src/sentry/api/endpoints/chunk_upload.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+
+# from rest_framework.permissions import IsAuthenticated
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry.models import FileBlob
+from sentry.models.file import DEFAULT_BLOB_SIZE
+from sentry.api.base import Endpoint
+from sentry.api.bases.project import ProjectReleasePermission
+
+
+UPLOAD_ENDPOINT = 'https://sentry.io'
+MAX_CHUNKS_PER_REQUEST = 16
+MAX_CONCURRENCY = 4
+HASH_ALGORITHM = 'sha1'
+
+
+class ChunkUploadEndpoint(Endpoint):
+    permission_classes = (ProjectReleasePermission, )
+
+    def get(self, request):
+        return Response(
+            {
+                'url': UPLOAD_ENDPOINT,
+                'chunkSize': DEFAULT_BLOB_SIZE,
+                'chunksPerRequest': MAX_CHUNKS_PER_REQUEST,
```

Comment:
Is there a reason to even accept multiple chunks per request? If we limited it to just 1 per request, you'd be able to achieve more throughput through parallelization. Anything in `chunksPerRequest` is going to be limited serially.

Reply:
The idea behind this was that if we send more (data) per request we save the overhead of new http connections.

Reply:
If we get sentry-cli to keep the connection alive, it should probably not even matter too much. But we will probably not implement parallel uploads in CLI...

Reply:
I'd argue that it'd be outweighed by the parallelism. For example, this is what `gsutil` does for it's own chunked uploads to Google Storage. I can't think of anything that does this hybrid thing where you can send chunks, but in batches. I don't have a strong opinion, but I don't think it's that valuable.

See stuff like: https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload

I also like the idea of using the `Content-Range` header, but might not be as relevant for us here since they are fixed sizes, etc.

But anyways, I'd really just vote for handling 1 chunk per upload to reduce some complexity.

Reply:
@mattrobenolt 1 chunk per upload will be awful for europeans. Means 1000 requests with a minimum latency of 60-150ms. We will be mostly waiting for network here.